File: UBB-CS/Year1/Sem2/Object_Oriented_Programming/Examples/TaskManager/Ui.cpp
#include "Ui.h"
#include <iostream>

using namespace std;

ui::ui(Services *srv) {
    this->srv = srv;
}

void ui::print_menu() {
    cout<<"1.Add a new task"<<endl;
    cout<<"2.Show all tasks"<<endl;
    cout<<"3.Show all tasks with less priority than selected"<<endl;
    cout<<"0.Exit"<<endl;
}

void ui::run() {
    while(true)
    {
        ui::print_menu();
        int command = { 0 };
        cout<<"Input command: ";
        cin>>command;
        cin.ignore();
        if(command == 0)
            break;
        else if(command == 1) {
            ui::add_task();
        }
        else if(command == 2){
            ui::show_tasks();
        }
        else if(command == 3){
            ui::show_filtered_tasks();
        }
        else
            cout<<"Wrong input"<<endl;
    }
}

void ui::add_task() {
    string desc;
    cout<<"Description: ";
    getline(cin,desc);
    int duration;
    cout<<"Duration: ";
    cin>>duration;
    cin.ignore();
    int priority;
    cout<<"Priority: ";
    cin>>priority;
    cin.ignore();

    int c = srv->add(desc,duration,priority);
    if(c==1)
        std::cout<<"Task succesfully added"<<std::endl;
    else if(c==0)
        cout<<"Task already exists"<<std::endl;
}

void ui::show_tasks() {
    Repository* repo = srv->getRepo();
    for(int i=0; i<repo->getSize(); i++)
        cout<<(*repo->getAllRepo())[i].toString()<<endl;
}

void ui::show_filtered_tasks() {
    int priority;
    cout<<"Priority: ";
    cin>>priority;
    cin.ignore();

    auto* valid_tasks = new task[srv->getRepo()->getSize()];
    int length = srv->FilteredTasks(valid_tasks, priority);
    if(length == 0)
        cout<<"No tasks match your criteria"<<endl;
    else
    {
        for(int j=0; j<length; j++)
            cout<<valid_tasks[j].toString()<<endl;
    }
}

File: UBB-CS/Year1/Sem1/Fundamentele Programarii/ExamPrep/Quizez/Interface/UI.py
class ui:

    def __init__(self, srv):
        self.srv = srv

    def run(self):
        while True:
            cmd = input("sex: ")
            cmd_name = cmd.split(' ')[0]
            if cmd_name == 'exit':
                return
            elif cmd_name == '':
                continue
            elif cmd_name == 'add':
                self.add(cmd[4:])
            elif cmd_name == 'create':
                self.create(cmd[7:])
            elif cmd_name == 'start':
                self.start(cmd[6:])
            else:
                print("please give valid command")

    def add(self, cmd):
        try:
            self.srv.add(cmd)
            print("added successfully")
        except ValueError:
            print("wrong input")
        except:
            print("id already exists")

    def create(self, cmd):
        try:
            self.srv.create(cmd)
        except ValueError:
            print("wrong input")

    def start(self, cmd):
        try:
            self.srv.start(cmd)
        except ValueError:
            print("wrong input")
